User: Added logout functionality

usmanasif opened the pull request:
Added `redis` package & a new environment variable `REDIS_URL` to connect with redis.

rweald reviewed (commented):
Why would we want to add the operational, security, and local development complexity of introducing a new database (Redis) rather than simply re-using the existing MongoDB database?

usmanasif commented:
@rweald i was using the practice of blacklisting token using redis as explained [here](https://stackoverflow.com/questions/37959945/how-to-destroy-jwt-tokens-on-logout) mainly due to its fast reads. However as you've suggested i should go with MongoDB instead so i'm thinking of adding `accessToken` in the user model & check it while authenticating the user. Please let me know if you've any concerns in this approach. I'll then implement it right away. Thanks

usmanasif commented:
@rweald i've added `accessToken` to `Coach` model. I've also limited the payload for generating the `accessToken` as it length was exceeding MongoDB's column limit. I've attached `coach` to `req` in `auth` middleware & i've updated the code where `userId` was being used. Please let me know if i've missed any case. Thanks
